ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/FileOperations.java

```diff
@@ -136,18 +137,41 @@ private void copyMmPath() throws LoginException, IOException {
     }
   }
 
-
-
   /**
    * This needs the root data directory to which the data needs to be exported to.
    * The data export here is a list of files either in table/partition that are written to the _files
    * in the exportRootDataDir provided.
    */
   private void exportFilesAsList() throws SemanticException, IOException {
-    // This is only called for replication that handles MM tables; no need for mmCtx.
-    try (BufferedWriter writer = writer()) {
-      for (Path dataPath : dataPathList) {
-        writeFilesList(listFilesInDir(dataPath), writer, AcidUtils.getAcidSubDir(dataPath));
+    if (dataPathList.isEmpty()) {
+      return;
+    }
+    boolean done = false;
+    int repeat = 0;
+    while (!done) {
+      // This is only called for replication that handles MM tables; no need for mmCtx.
+      try (BufferedWriter writer = writer()) {
+        for (Path dataPath : dataPathList) {
+          writeFilesList(listFilesInDir(dataPath), writer, AcidUtils.getAcidSubDir(dataPath));
+        }
+        done = true;
+      } catch (IOException e) {
+        repeat++;
+        logger.info("writeFilesList failed with exception : " + e.getMessage() + " for " + repeat + " time");
```

Comment:
This log can be added for each attempt before copy so that even for success case, we will know the list of files.

Reply:
i am not sure if we should add info logs for success case